functions/modelPreparation.R

```diff
@@ -46,6 +46,26 @@ modelPreparation <- function(focalTaxa, speciesData, redListModelled, regionGeom
     })
     focalSpeciesDataRefined <- focalSpeciesDataRefined[!is.na(focalSpeciesDataRefined)]
     
+    
+    # Combine species by functionalGroup if requested (else leave as separate)
+    # identify species with data
+    uniqueTaxaSpecies <- unique(unlist(lapply(focalSpeciesDataRefined, function(ds){
+      as.character(ds$acceptedScientificName)
+    })))
+    
+    # identify functional groups in species with data for focal taxonomic group
+    focalSpeciesWithData <- focalTaxa[focalTaxa$key %in% uniqueTaxaSpecies &  # species with data
```

Comment:
Doesn't work for me. focalTaxa is a vector, yet here we call it like a dataframe.

Reply:
Yeah so this bit needs to be the full focalTaxon table, not the focalTaxa vector. What we can do is change speciesModelRuns to instead import the table and filter the table as opposed to turning focalTaxa into a filtered vector.

Reply:
Fixed this. uniqueTaxaSpecies should be based on taxonKeyProject, not acceptedScientificName. fixed below.